apisix/core/ctx.lua

```diff
@@ -215,11 +215,18 @@ do
                 key = sub_str(key, 9)
                 val = get_parsed_graphql()[key]
 
-            elseif apisix_var_names[key] then
-                val = ngx.ctx.api_ctx and ngx.ctx.api_ctx[key]
-
             else
-                val = get_var(key, t._request)
+                local getter = apisix_var_names[key]
+                if getter then
+                    if getter == true then
+                        val = ngx.ctx.api_ctx and ngx.ctx.api_ctx[key]
+                    else
+                        val = getter(ngx.ctx.api_ctx)
```

Comment:
Maybe we should add a comment here, to notify developer where the `getter` from ?

Reply:
@leslie-tsang
Updated.